### PR TITLE
Tracks: Sanitize error messages sent to tracks.

### PR DESCRIPTION
**Pull request by aerych**
For background see p4qSXL-2xV-p2

In certain scenarios we were sending the entire http response from a failed xmlrpc request to tracks as part of login error tracking.   This pr checks for the offending content and truncates it to preserve context, but avoid sending unhelpful info to tracks.

To test:
For a practical test you could edit a self-hosted test blog's xmlrpc.php file so it yields an error.  Set break points to inspect the diff then attempt to login to a self-hosted site. 

@astralbodies would you kindly?

### CODE REVIEW

File: WordPress/Classes/Utility/Analytics/WPAppAnalytics.m [hunk 1]
```diff
@@ -276,14 +277,34 @@ + (void)track:(WPAnalyticsStat)stat withProperties:(NSDictionary *)properties {
...
+                                 @"error_domain": err.domain,
+                                 @"error_description": err.description
     };
     [self track:stat withProperties: properties];
 }
 
+/**
+ * @brief   Sanitize an NSError so we're not tracking unnecessary or usless information.
+ */
++ (NSError *)sanitizedErrorFromError:(NSError * _Nonnull)error
```
Comment (astralbodies) [inline WordPress/Classes/Utility/Analytics/WPAppAnalytics.m]: Should this return a `_Nonnull` error as well?

**Comment by aerych** [comment]
Thanks @astralbodies :)  Updated.

**Comment by astralbodies** [comment]
:shipit:

**Comment by aerych** [comment]
Thank ya sir!

**Pull request merged**